pandas/core/arrays/base.py

```diff
@@ -592,7 +592,17 @@ def to_numpy(
         -------
         numpy.ndarray
         """
-        result = np.asarray(self, dtype=dtype)
+        if dtype == "datetime64" and self.dtype.kind == "M":
```

Comment:
This fix is too specific. I think the `TypeError` raised in the original issue should be root caused first.